### ISSUE TITLE
flag esc_js inside <script> tag

### ISSUE THREAD

**Issue by david-binda** [Automattic/VIP-Coding-Standards#113]
It's almost always meant to be `wp_json_encode`

**Comment by david-binda** [comment]
We could check via sniff whether there is an open HTML attribute prior `esc_js`, and if not, it's in 99% of cases wrongly used function.

**Comment by GaryJones** [comment]
Please provide some more context here, such as example code that should be reported, and example code that shouldn't.

**Comment by GaryJones** [comment]
Is this something that would benefit everyone? i.e. not VIP-specific?

Seems like https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/master/WordPress/Sniffs/Security/EscapeOutputSniff.php would be a good location for this to be addressed in?

**Comment by rebeccahum** [comment]
Also other escaping functions like `esc_attr` or `esc_url`:

```
<script type='text/javascript'>

somefunction('somefuncprefix_<?php echo esc_attr( $myid ); ?>_div').setup({
   file: '<?php echo esc_url( $myurl ); ?>',
})

</script>
```

**Comment by GaryJones** [comment]
While `esc_js()` is almost certainly wrong, I think we'd be better pointing to https://github.com/Automattic/vip-code-samples/tree/master/10-security#escaping-dynamic-javascript-values or similar and letting the developer work out what the correct cod should b.

**Comment by rebeccahum** [comment]
@GaryJones Agreed, since context is important in this aspect.  However, I think flagging `esc_js()` usage would be a valid sniff.
